var/spack/repos/builtin/packages/py-jupyterlab/package.py

```diff
@@ -14,15 +14,18 @@ class PyJupyterlab(PythonPackage):
     # Skip 'jupyterlab.tests' packages
     import_modules = ['jupyterlab', 'jupyterlab.handlers']
 
+    version('3.1.14', sha256='13174cb6076dd5da6f1b85725ccfcc9518d8f98e86b8b644fc89b1dfaeda63a9')
+    version('3.0.18', sha256='0e4bb4b89014607a16658b54f13df2f0af14f3c286109a0e14d5a46cbbe28caf')
     version('3.0.16', sha256='7ad4fbe1f6d38255869410fd151a8b15692a663ca97c0a8146b3f5c40e275c23')
+    version('3.0.14', sha256='713a84991dfcca8c0bc260911f1bd54ac25a386a86285713b9555a60f795059b')
     version('2.2.7', sha256='a72ffd0d919cba03a5ef8422bc92c3332a957ff97b0490494209c83ad93826da')
     version('2.1.0', sha256='8c239aababf5baa0b3d36e375fddeb9fd96f3a9a24a8cda098d6a414f5bbdc81')
 
     depends_on('python@3.6:', when='@3:', type=('build', 'run'))
     depends_on('python@3.5:', type=('build', 'run'))
     depends_on('py-setuptools', type=('build', 'run'))
     depends_on('py-jupyter-packaging@0.9:1.999', when='@3.0.15:', type='build')
-    depends_on('py-jupyter-packaging@0.7.3:0.7.999', when='@3.0.0:3.0.14', type='build')
+    depends_on('py-jupyter-packaging@0.7.3:0.7.999', when='@3.0.0:3.0.14', type=('build', 'run'))
```

Comment:
According to the `setup.py` this should actually be `0.7:0.999` right? Are you sure it's actually needed at run-time? I don't see it imported anywhere other than in `setup.py`.

Reply:
Yes, according to `setup.py` it should be what you suggested, but in [pyproject.toml](https://github.com/jupyterlab/jupyterlab/blob/v3.0.14/pyproject.toml#L2) they restrict the version to `~=0.7.3`. That is why I also used that one.

Regarding run-time usage, I have added it because till `py-jupyterlab` 3.0.14 `py-jupyter-packaging` is in install_requires. Therefore when I tried to run it, jupyterlab failed to start up with the error message:
```
pkg_resources.DistributionNotFound: The 'jupyter_packaging~=0.7' distribution was not found and is required by jupyterlab
```
Adding it as run-time dependency solved that.